napari/layers/tracks/_track_utils.py

```diff
@@ -306,6 +306,26 @@ def build_tracks(self):
         self._track_vertices = np.concatenate(track_vertices, axis=0)
         self._track_connex = np.concatenate(track_connex, axis=0)
 
+    def build_tracks(self):
+        """build the tracks"""
+
+        # Track ids associated to all vertices, sorted by time
+        points_id = self.data[:, 0][self._ordered_points_idx].copy()
+        # Coordinates of all vertices
+        track_vertices = self.data[:, 1:].copy()
```

Comment:
is `copy()` actually necessary here? As far as I can tell, these arrays are never modified inplace up until they're given to vispy:

- `points_id` is only indexed via `get_value`
- `track_vertices` is returned via `_view_data`

Sometimes vispy does change arrays, so I'm not 100% sure it's fine to remove these, but since we're here maybe we can test a bit!

Reply:
I tend to put `.copy()` instinctively now (had _many_ bad surprises in the past). Any idea how to make tests to check whether or not they are modified later down the visualization line ?

Reply:
I took a look into vispy and I'm 90% sure this `copy` is unneeded. I'm not sure how to test thoroughly... some things that cause issues in the past are: passing dtypes that need conversion (float64) with data outside of the range of float32; changing data inplace on the napari side (layer.data[something] = something), then triggering a `layer.refresh()`... and seeing if things go wrong ^^'